src/main/java/dk/xam/jbang/Main.java

```diff
@@ -260,6 +260,42 @@ public Integer call() throws IOException {
 		}
 	}
 
+	static boolean checkVersions() {
+		String version = System.getProperty("java.version");
+		if (version.startsWith("1.")) {
+			version = version.substring(2, 3);
+		} else {
+			int dot = version.indexOf(".");
+			if (dot != -1) {
+				version = version.substring(0, dot);
+			}
+		}
+		int javaVersion = Integer.parseInt(version);
+		int javacVersion = 0;
+		try {
+			Process p = Runtime.getRuntime().exec("javac -version");
+			StringBuilder output = new StringBuilder();
+			BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()));
+			String line;
+			while ((line = reader.readLine()) != null) {
+				output.append(line + "\n");
+			}
+			String[] s = output.toString().split(" ");
+			if (s[1].startsWith("1.")) {
+				s[1] = s[1].substring(2, 3);
+			} else {
+				int dot = s[1].indexOf(".");
+				if (dot != -1) {
+					s[1] = s[1].substring(0, dot);
+				}
+			}
+			javacVersion = Integer.parseInt(s[1]);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		return (javaVersion == javacVersion);
```

Comment:
here you are comparing the java version jbang is running - with the javacVersion found in path. That is not the correct comparison.

what needs to be tested is if the java used for building and javac used for compiling are compatible.

Reply:
I am sorry to be repetitive but I am having a hard time figuring out how to resolve this.

Reply:
you can't just take the java path from the vm running `jbang` you need to call out to `java -version ` similar as you do for `javac`.

Reply:
Thanks a lot for this tip i implemented the call for `java -version` the same way for `javac -version`